arangod/Aql/ShardLocking.cpp

```diff
@@ -248,6 +248,7 @@ std::vector<ServerID> ShardLocking::getRelevantServers() {
                    TRI_ASSERT(!item.first.empty());
                    return item.first;
                  });
+  std::sort(result.begin(), result.end());
```

Comment:
I think this is a very important part of the fix.

Reply:
Can you please add a code comment on this function and in the header file that the Result is sorted, and that this sorting is required to avoid distributed dead-locking.
This increases the chance to not lose the sortedness in the future.

Reply:
ok now I am bit unsure if the sort here is actually required.
In the critical locking path (in ClusterTransaction state) the servers are sorted and contacted in serialized ordering.
Do both places need to be sorted?

Reply:
This is used in the AQL setup code